src/components/RoadTabs.vue

```diff
@@ -149,16 +161,22 @@ export default {
     createRoad: function () {
       if (!this.duplicateRoad) {
         this.$emit('add-road', this.newRoadName);
+        this.addDialog = false;
+        this.newRoadName = '';
       } else if (this.duplicateRoadSource in this.roads) {
-        this.$emit('add-road',
-          this.newRoadName,
-          this.roads[this.duplicateRoadSource].contents.coursesOfStudy.slice(0),
-          this.roads[this.duplicateRoadSource].contents.selectedSubjects.map((semester) => semester.slice(0)),
-          Object.assign({}, this.roads[this.duplicateRoadSource].contents.progressOverrides)
-        );
+        if (this.$store.state.unretrieved.indexOf(this.duplicateRoadSource) >= 0) {
+          this.$emit('retrieve', this.duplicateRoadSource);
+        } else {
+          this.$emit('add-road',
+            this.newRoadName,
+            this.roads[this.duplicateRoadSource].contents.coursesOfStudy.slice(0),
+            this.roads[this.duplicateRoadSource].contents.selectedSubjects.map((semester) => semester.slice(0)),
+            Object.assign({}, this.roads[this.duplicateRoadSource].contents.progressOverrides)
```

Comment:
seems like an exact copy of the `$store.state.unretrieved` watcher. is there a nice way to combine them?